test/utils/completable_app_widget_harness.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:async';
```

Comment:
This file has three classes that are variants of the same thing - a test harness that is meant to wrap a CrowdLeagueApp widget and allow external completing of the firebase and redux futures. 

I imagine there's a much nicer way to do this without three separate classes that do very similar things but I tried playing with inheritance and wasn't getting anywhere so I went this way for now and I figure anyone that wants to have a go at some point can try and improve the design.